### PR TITLE
Reviews V1

### PR DESCRIPTION
**Pull request by Raulido**
Completed the Tenant side of writing a Review. 

Summary of what was done:
Completed jobs now have a review button that opens a bootstrap form when clicked. Tenants can submit the form and upon submission a new review object with a belongs_to association with jobs is created with the fields from form. 

What still needs to be done:
Prevent multiple reviews per job.
Make a rating field for vendors.
Calculate ratings for vendors using all reviews associated with that vendor's job.

### CODE REVIEW

File: app/javascript/components/CalendarIndex.js [hunk 1]
```diff
@@ -97,6 +103,61 @@ const handleCompleteJob = (event, job_id) => {
...
 
+
+const handleUpdateVendorRating = (vendor_id, rate) => {
+  fetch(`http://localhost:3000/vendors/update_rating`, {
+    method: "PATCH",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({
+      vendor_id: vendor_id,
+      rating: rate,
+      num: 1
```
Comment (franklee26) [inline app/javascript/components/CalendarIndex.js]: Why is num being passed into the patch?

File: app/javascript/components/CalendarIndex.js [hunk 2]
```diff
@@ -97,6 +103,61 @@ const handleCompleteJob = (event, job_id) => {
...
+        alert("Failed to update vendor");
+      }
+    });
+};
+
+const handleJobReview = (event, job_id, text, rate, vendor_id) => {
+  console.log(job_id);
+  console.log(vendor_id);
+  console.log(text);
+  console.log(rate);
```
Comment (franklee26) [inline app/javascript/components/CalendarIndex.js]: console logs here

File: app/controllers/vendors_controller.rb [hunk 3]
```diff
@@ -61,6 +61,30 @@ def update_vendor
...
+
+def update_rating
+  body = JSON(request.body.read)
+  rating = body["rating"]
+  vendor_id = body["vendor_id"]
+  num = body["num"]
+  vendor = Vendor.find_by(id: vendor_id)
+  if vendor 
+    if vendor.rating
+      r = (vendor.rating).to_f * (vendor.num).to_f * 1.0 + (rating).to_f * 1.0
```
Comment (franklee26) [inline app/controllers/vendors_controller.rb]: nice, this looks like a great way to cumulate ratings

File: app/javascript/components/CalendarIndex.js (hunk 1 above)
Comment (Raulido) [reply app/javascript/components/CalendarIndex.js]: No reason. I'll take it off.

File: app/javascript/components/CalendarIndex.js (hunk 2 above)
Comment (Raulido) [reply app/javascript/components/CalendarIndex.js]: Debugging was interesting.

File: app/controllers/vendors_controller.rb (hunk 3 above)
Comment (Raulido) [reply app/controllers/vendors_controller.rb]: TY

**Pull request merged**